lib/carto/renderer_js.js

```diff
@@ -244,6 +238,13 @@ CartoCSS.prototype = {
       var done = {};
       for(var i = 0; i < defs.length; ++i) {
         var def = defs[i];
+
+        def.toXML(parse_env, {});
```

Comment:
At the definition level they are completely built with the toJS() method, if you use toXML at rule level you're gonna have a type error, taking all the rules as expressions.